spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -727,51 +740,97 @@ def write_sorting(sorting, save_path, **nwbfile_kwargs):
                     spkt = sorting.get_unit_spike_train(unit_id=id) / fs # spike trains withinin the SortingExtractor object are not scaled by sampling frequency
                     nwbfile.add_unit(id=id, spike_times=spkt)
 
+            # Always add column for shank ids if a MultiSortingExtractor
+            if isinstance(sorting,se.MultiSortingExtractor):
+                set_dynamic_table_property(
+                        dynamic_table=nwbfile.units,
+                        row_ids=[int(k) for k, v in sorting._unit_map.items()],
+                        property_name='shank_id',
+                        values=[unit_dict['unit_id']-1 for ind,unit_dict in sorting._unit_map.items()],
+                        default_value=np.nan,
+                        description='0-indexed id of cluster of shank'
+                )
+
             # Units properties
             for pr in all_properties:
-                unit_ids = [int(k) for k, v in sorting._properties.items()
+                unit_ids = [int(k) for k, v in total_prop_dict.items()
                             if pr in v]
-                vals = [v[pr] for k, v in sorting._properties.items()
-                        if pr in v]
+                
+                # We can always guarantee two levelx of nested dictionaries based
+                # based on the current structure
+                # If the value inside that is also a dictionary, require that 
+                # it is of the {'name': my_name, 'description':my_description, 'data':my_data} format
+                unit_prop = next(iter(next(iter(total_prop_dict.values())).values()))
+                if type(unit_prop) is dict:
+                    item_names = [x for x in unit_prop.keys()]
+                    
+                    if 'name' not in item_names:
+                        name = pr
+                    else:
+                        name = [v[pr]['name'] for k, v in total_prop_dict.items()][0]
+                        
+                    if 'description' not in item_names:
+                        description = 'no description'
+                    else:
+                        description = [v[pr]['description'] for k, v in total_prop_dict.items()][0]
+                    
+                    if 'data' not in item_names:
+                        assert False, 'If setting a unit_property to be a dictionary, ensure a "data" field exists!'
+                    else:
+                        vals = [v[pr]['data'] for k, v in total_prop_dict.items()]
+                else:
+                    name = pr
+                    vals = [v[pr] for k, v in total_prop_dict.items()
+                            if pr in v]
+                    description = 'no description'
+                
+                # Special case of setting max_electrodes requires a table to be
+                # passed to become a dynamic table region
+                if name == 'max_electrode':
```

Comment:
this is `max_channel` in SI usually

Reply:
Can you provide a source/citation for this? It is current referred to as `max_electrode` on the current conversions (and formats currently on DANDI), so we would want to ensure that convention is correct.